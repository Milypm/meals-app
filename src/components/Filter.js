import PropTypes from 'prop-types';
import '../styles/navbar.css';

const Filter = (props) => {
  const { onChangeHandler, currentCateg } = props;
  const categoryFilter = [
    'All',
    'Beef',
    'Breakfast',
    'Chicken',
    'Dessert',
    'Goat',
    'Lamb',
    'Miscellaneous',
    'Pasta',
    'Pork',
    'Seafood',
    'Side',
    'Starter',
    'Vegan',
    'Vegetarian',
  ];
  return (
    <form className="filter-form">
      <label className="label-filter">Category:</label>
      <select className="select-filter" id="category" name="category" onChange={(e) => onChangeHandler(e)} value={currentCateg}>
        {
          categoryFilter.map((cat) => (
            <option key={cat} className="option-filter" value={cat}>
              {cat}
            </option>
          ))
        }
      </select>
    </form>
  );
};
Filter.defaultProps = {
  onChangeHandler: () => {},
  currentCateg: '',
};
Filter.propTypes = {
  onChangeHandler: PropTypes.func,
  currentCateg: PropTypes.string,
};
export default Filter;
