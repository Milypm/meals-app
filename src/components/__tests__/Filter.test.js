import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Filter from '../Filter';

describe('Filter component', () => {
  test('Filter component renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Filter currentCateg="All" />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
