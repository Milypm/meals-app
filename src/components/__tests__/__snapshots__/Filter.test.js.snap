// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component Filter component renders correctly 1`] = `
<form
  className="filter-form"
>
  <label
    className="label-filter"
  >
    Category:
  </label>
  <select
    className="select-filter"
    id="category"
    name="category"
    onChange={[Function]}
    value="All"
  >
    <option
      className="option-filter"
      value="All"
    >
      All
    </option>
    <option
      className="option-filter"
      value="Beef"
    >
      Beef
    </option>
    <option
      className="option-filter"
      value="Breakfast"
    >
      Breakfast
    </option>
    <option
      className="option-filter"
      value="Chicken"
    >
      Chicken
    </option>
    <option
      className="option-filter"
      value="Dessert"
    >
      Dessert
    </option>
    <option
      className="option-filter"
      value="Goat"
    >
      Goat
    </option>
    <option
      className="option-filter"
      value="Lamb"
    >
      Lamb
    </option>
    <option
      className="option-filter"
      value="Miscellaneous"
    >
      Miscellaneous
    </option>
    <option
      className="option-filter"
      value="Pasta"
    >
      Pasta
    </option>
    <option
      className="option-filter"
      value="Pork"
    >
      Pork
    </option>
    <option
      className="option-filter"
      value="Seafood"
    >
      Seafood
    </option>
    <option
      className="option-filter"
      value="Side"
    >
      Side
    </option>
    <option
      className="option-filter"
      value="Starter"
    >
      Starter
    </option>
    <option
      className="option-filter"
      value="Vegan"
    >
      Vegan
    </option>
    <option
      className="option-filter"
      value="Vegetarian"
    >
      Vegetarian
    </option>
  </select>
</form>
`;
